cmd/fleet/serve.go

```diff
@@ -1153,22 +1152,23 @@ the way that the Fleet server works.
 
 				if (req.Method == http.MethodPost && strings.HasSuffix(req.URL.Path, "/fleet/software/package")) ||
 					(req.Method == http.MethodPatch && strings.HasSuffix(req.URL.Path, "/package") && strings.Contains(req.URL.Path, "/fleet/software/titles/")) {
-					// when uploading a software installer, the file might be large so
-					// the read timeout (to read the full request body) must be extended.
+					var zeroTime time.Time
 					rc := http.NewResponseController(rw)
-					// the frontend times out waiting for the upload after 4 minutes,
-					// use that same timeout:
-					// https://www.figma.com/design/oQl2oQUG0iRkUy0YOxc307/%2314921-Deploy-security-agents-to-macOS%2C-Windows%2C-and-Linux-hosts?node-id=773-18032&t=QjEU6tc73tddNSqn-0
-					if err := rc.SetReadDeadline(time.Now().Add(softwareInstallerUploadTimeout)); err != nil {
+					// For large software installers, the server time needs time to read the full
+					// request body so we use the zero value to remove the deadline and override the
+					// default read timeout.
+					// TODO: Is this really how we want to handle this? Or would an arbitrarily long
+					// timeout be better?
```

Comment:
For Fleet-maintained apps, we're using a 15-minute [timeout](https://github.com/fleetdm/fleet/blob/81a95a742601bbd03997f82e52aa9a8b8e0fc4e5/server/mdm/maintainedapps/installers.go#L18-L19)